pulser/tests/test_simulation.py

```diff
@@ -307,3 +334,57 @@ def test_run():
                                    np.array([0, sim._times[-10],
                                              sim._times[-3], sim._times[-1]])
                                    )
+
+
+def test_get_xy_hamiltonian():
+    simple_reg = Register.from_coordinates(
+        [[0, 10], [10, 0], [0, 0]], prefix='atom')
+    detun = 1.
+    amp = 3.
+    rise = Pulse.ConstantPulse(1500, amp, detun, 0.)
+    simple_seq = Sequence(simple_reg, MockDevice)
+    simple_seq.declare_channel('ch0', 'mw_global')
+    simple_seq.set_magnetic_field(0, 1., 0.)
+    simple_seq.add(rise, 'ch0')
+
+    assert np.abs(np.linalg.norm(simple_seq.magnetic_field[0:2]) - 1) < 1e-10
```

Comment:
Just a tip: You can use `np.isclose()` for these comparisons

Reply:
How do you think it should be ?

Reply:
It would be 
```suggestion
    assert np.isclose(np.linalg.norm(simple_seq.magnetic_field[0:2]), 1)
```
You can optionally define the tolerance if you need to.

Reply:
I was asking about your previous comment above actually

Reply:
About the previous comment, I don't see a reason for it to change, but I might be missing something. I would revert to the previous version and check if everything still runs well.